2.4/root/usr/share/container-scripts/httpd/common.sh

```diff
@@ -90,3 +90,41 @@ generate_container_user() {
   export NSS_WRAPPER_GROUP=/etc/group
 }
 
+# Copy config files from application to the location where httd expects them
+# Param sets the directory where to look for files
+process_config_files() {
+  local dir=${1:-.}
+  if [ -d ${dir}/httpd-cfg ]; then
+    echo "---> Copying httpd configuration files..."
+    if [ "$(ls -A ${dir}/httpd-cfg/*.conf)" ]; then
+      cp -v ${dir}/httpd-cfg/*.conf "${HTTPD_CONFIGURATION_PATH}"
+      rm -rf ${dir}/httpd-cfg
+    fi
+  else
+    if [ -d ${dir}/cfg ]; then
+      echo "---> Copying httpd configuration files from deprecated './cfg' directory, use './httpd-cfg' instead..."
+      if [ "$(ls -A ${dir}/cfg/*.conf)" ]; then
+        cp -v ${dir}/cfg/*.conf "${HTTPD_CONFIGURATION_PATH}"
+        rm -rf ${dir}/cfg
+      fi
+    fi
+  fi
+}
+
+# Copy SSL files provided in application source
+process_ssl_certs() {
+  local dir=${1:-.}
+  if [ -d ${dir}/httpd-ssl/private ] && [ -d ${dir}/httpd-ssl/certs ]; then
+    echo "---> Looking for SSL certs for httpd..."
+    cp -r ${dir}/httpd-ssl ${HTTPD_APP_ROOT}
+    local ssl_cert="$(ls -A ${HTTPD_APP_ROOT}/httpd-ssl/certs/*.pem | head -n 1)"
+    local ssl_private="$(ls -A ${HTTPD_APP_ROOT}/httpd-ssl/private/*.pem | head -n 1)"
+    if [ -f "${ssl_cert}" ] && [ -f "${ssl_private}" ]; then
+      echo "---> Setting SSL certs for httpd..."
+      sed -i -e "s|^SSLCertificateFile .*$|SSLCertificateFile ${ssl_cert}|" ${HTTPD_MAIN_CONF_D_PATH}/ssl.conf
+      sed -i -e "s|^SSLCertificateKeyFile .*$|SSLCertificateKeyFile ${ssl_private}|" ${HTTPD_MAIN_CONF_D_PATH}/ssl.conf
```

Comment:
So keeping this branch but adding an
```
elif [ -f "${ssl_cert}" ]
  # do sed for SSLCertificateFile
  sed -i /SSLCertificateKeyFile .*/d  ${HTTPD_MAIN_CONF_D_PATH}/ssl.conf
fi
```
is probably right, i.e. remove SSLCertificateKeyFile if *only* a cert file is present.

Reply:
Thanks for this suggestion, I'll change it this way.